backend/compact-connect/lambdas/python/common/cc_common/data_model/data_client.py

```diff
@@ -608,6 +608,17 @@ def _rollback_privilege_transactions(
                 raise CCAwsServiceException('Failed to roll back privilege transactions') from e
         logger.info('Privilege rollback complete')
 
+    def _get_all_military_affiliation_records_for_provider(self, compact: str, provider_id: str):
+        military_affiliation_records = self.config.provider_table.query(
+            KeyConditionExpression=Key('pk').eq(f'{compact}#PROVIDER#{provider_id}')
+            & Key('sk').begins_with(
+                f'{compact}#PROVIDER#military-affiliation#',
+            ),
+        ).get('Items', [])
+
+        schema = MilitaryAffiliationRecordSchema()
+        return [schema.load(record) for record in military_affiliation_records]
```

Comment:
Nit - we are attempting to move towards using the data class pattern for all of our serializing/deserializing rather than referencing the marshmallow schemas directly. We have a data class for military affiliation records that you should be able to leverage:
```suggestion
        return [MilitaryAffiliationData.from_database_record(record) for record in military_affiliation_records]
```
This would mean you are now returning the data class instances, rather than dictionaries. So the 'inactivate_military_affiliation_status' method would need to be updated to account for this. I will show an example for this in the next comment.